xmake/rules/c++/modules/modules_support/msvc.lua

```diff
@@ -181,16 +192,20 @@ function generate_stl_headerunits_for_batchjobs(target, batchjobs, headerunits,
         if not os.isfile(bmifile) or not os.isfile(objectfile) then
             batchjobs:addjob(headerunit.name, function(index, total)
                depend.on_changed(function()
-                    progress.show((index * 100) / total, "${color.build.object}generating.cxx.headerunit.bmi %s", headerunit.name)
-                    local args = {headernameflag .. ":angle", headerunit.name, ifcoutputflag, stlcachedir, "-Fo" .. objectfile}
-                    os.vrunv(compinst:program(), table.join(compinst:compflags({target = target}), common_args, args), {envs = vcvars})
+                    if not common.localcache():get2(headerunit.name, "building") then 
```

Comment:
why not using memcache() or `_g.xxx` to cache build state? localcache will write/read files.